src/cosl/coordinated_workers/nginx.py

```diff
@@ -29,6 +31,425 @@
     "nginx_port": 8080,
     "nginx_exporter_port": 9113,
 }
+RESOLV_CONF_PATH = "/etc/resolv.conf"
+
+
+class NginxLocationModifier(str, Enum):
+    """Enum representing valid Nginx `location` block modifiers."""
+
+    # cfr. https://www.digitalocean.com/community/tutorials/nginx-location-directive#nginx-location-directive-syntax
+
+    PREFIX = ""  # prefix match
+    EXACT = "="  # exact match
+    REGEX_CASE_SENSITIVE = "~"  # case-sensitive regex match
+    REGEX_CASE_INSENSITIVE = "~*"  # case-insensitive regex match
+    PREFIX_NO_REGEX = "^~"  # prefix match that disables further regex matching
+
+
+@dataclass
+class NginxLocationConfig:
+    """Represents a `location` block in an Nginx configuration.
+
+    For example, NginxLocationConfig('/', 'foo', backend_url="/api/v1" headers={'a': 'b'}, modifier=EXACT, is_grpc=True)
+    would result in:
+        location = / {
+            set $backend grpc://foo/api/v1;
+            grpc_pass $backend;
+            proxy_connect_timeout 5s;
+            proxy_set_header a b;
+        }
```

Comment:
Shouldn't we have a docstring in this file that explains the usage of the objects provided by this lib?

For instance, we have that in our charm libs: https://github.com/canonical/loki-k8s-operator/blob/main/lib/charms/loki_k8s/v1/loki_push_api.py#L7